src/lm_buddy/jobs/_entrypoints/finetuning.py

```diff
@@ -101,14 +106,31 @@ def training_function(config_data: dict[str, Any]):
     print(f"Training result: {result}")
 
     # Register a model artifact if tracking is enabled and Ray saved a checkpoint
-    if config.tracking and result.checkpoint:
-        # Must resume from the just-completed training run
-        with wandb_init_from_config(config.tracking, resume=WandbResumeMode.MUST) as run:
-            model_artifact = build_directory_artifact(
-                artifact_name=default_artifact_name(run.name, ArtifactType.MODEL),
-                artifact_type=ArtifactType.MODEL,
-                dir_path=f"{result.checkpoint.path}/{RayTrainReportCallback.CHECKPOINT_NAME}",
-                reference=True,
-            )
-            print("Logging artifact for model checkpoint...")
-            artifact_loader.log_artifact(model_artifact)
+    ckpt_path, artifact_config = None, None
```

Comment:
In a follow-up PR (https://mzai.atlassian.net/browse/RD2024-152), I would like to refactor a bit how we are generating artifacts and results in these methods. 

The issue is that because the tracking field is optional, we repeatedly end up having to write two code branches, (1) for when we initialize a W&B run and create an artifact and (2) for when we run the job without tracking or artifacts. It's almost like we want something like `maybe_initialize_wandb_run` that handles the optionality of the tracking service.